src/openassetio-python/package/openassetio/test/manager/apiComplianceSuite.py

```diff
@@ -729,6 +731,8 @@ def test_when_querying_malformed_reference_then_malformed_reference_error_is_ret
     def __testGetWithRelationships(
         self, reference, relationships, expected_relations, resultTraitSet=None
     ):
+        if resultTraitSet is None:
+            resultTraitSet = set()
```

Comment:
[question] Is this a pybind thing? ie. setting a kwarg to `None` is treated differently to not setting it at all?

Reply:
[plan] I knew there was something in there somewhere. Not related to this PR per-se, but we should double check`py::arg("resultTraitSet") = trait::TraitSet{})` doesn't risk the same problems as if you did that in a `def`, and fix if so.

Reply:
Added https://github.com/OpenAssetIO/OpenAssetIO/issues/987 - but, as noted in the ticket, it looks like there's no problem.